docs/cli.rst

```diff
@@ -1265,6 +1265,12 @@ facilitates rapid iterations, inspections of any pipeline as well as debugging.
     $ nextflow run nextflow-io/hello -with-tower
  
 
+- Invoke the nextflow pipeline execution with a custom parameters ``YML`` file. 
+The parameters which are specified through this mechanism are merged with the resolved configuration (base configuration and profiles) and only the common fields are overwritten by the ``YML`` file.::
```

Comment:
The params file can be YAML or JSON. I would say YAML/JSON file instead of `YML` file.

Reply:
Also @pditommaso we should clarify what happens if you provide params through a config file and a params file via CLI arguments. I think the params file takes priority but I'm not certain.

Reply:
Yes, this question has come up recently https://github.com/nextflow-io/nextflow/issues/2662

Reply:
Was missing this conversation. The CLI params overrides the ones the json/yaml file.

You can see it here, it reads the params file and put it into the `result` map. then if there are cli `params` copy them over the `result` map. 

https://github.com/nextflow-io/nextflow/blob/0f63f1a7cc7953ed1d014e8d3f821ba2941c6177/modules/nextflow/src/main/groovy/nextflow/cli/CmdRun.groovy#L496-L513